test/CMakeLists.txt

```diff
@@ -25,7 +25,13 @@
 cmake_policy(SET CMP0057 NEW)
 
 find_package(Threads REQUIRED)
-rocm_test_link_libraries(Threads::Threads migraphx migraphx_onnx migraphx_tf)
+rocm_test_link_libraries(Threads::Threads migraphx)
+if(MIGRAPHX_ENABLE_TENSORFLOW)
+    rocm_test_link_libraries(migraphx_tf)
+endif()
+if(MIGRAPHX_ENABLE_ONNX)
+    rocm_test_link_libraries(migraphx_onnx)
```

Comment:
This is why the ref tests work for onnx.hpp as we are linking onnx for everything. Just a note, we probably need to fix that so we can check for erroneous includes in our CI, but not in this PR.